src/common/utils2.cpp

```diff
@@ -0,0 +1,69 @@
+#include "utils2.hpp"
+
+#include <chrono>
+#include <string>
+#include <algorithm>
+#include <iostream>
+#include <numeric> //iota
+
+
+using Clock = std::chrono::steady_clock;
+using std::chrono::time_point;
+using std::chrono::duration_cast;
+using std::chrono::microseconds;
+
+struct cScopeTimer::sPimpl {
```

Comment:
What's this used for?
Could we keep it out as long as we do not need it yet?

Reply:
http://en.cppreference.com/w/cpp/language/pimpl

Looks like it's quite useful in large libraries. I don't think we really need it in our small project.

Reply:
tough it might be usefull. why you guy not liking it ?
what the matter with it ?

Reply:
I wouldn't say I don't like it. It just feels hacky/dirty. Maybe it could cause issues in future versions. Maybe it could become a part of C++ itself with a single statement.
But most importantly: Why should we have it if we don't need it?

Reply:
hacky ? uh =(, (I just didn't want to expose it to avoid the need of othrer class to know full std::chrono and stuff implementation.
Why do we need it?
1) Well in debug mode using it would be perfectly valid. To know how long that suff take etc... The call could just be wrapped with some macro to disable. (in release you could too but it's less common as you want the full speed..)
2) I view it as a tool. Since I used firewall / pimpl template. the compilation time won't be altered, all that we really losing is on the .exe since. Not that important for me atm.
3) Basic_test or some other will need it, plus I think it's good to have an example of how to play with those timer.
For example I didn't know it would have fail with this implementation on vs. (I tryed with gcc as the comment say)
If next version of c++ have it one line that good, but I'm not sure will be able to use it right away. (gcc/clang/vs..)